continuous_integration/environment-3.10-dev.yaml

```diff
@@ -3,15 +3,15 @@ channels:
 - conda-forge
 - nodefaults
 dependencies:
-- c-compiler
+# - c-compiler
 - dask>=2022.3.0
 - fastapi>=0.92.0
 - fugue>=0.7.3
 - httpx>=0.24.1
 - intake>=0.6.0
 - jsonschema
 - lightgbm
-- maturin>=1.1,<1.2
+- maturin>=1.3.1
```

Comment:
There are a couple more files we should update this dep in:

- pyproject.toml
- docker/conda.txt
- docker/main.dockerfile
- docs/requirements-docs.txt

Reply:
Noted and done. Will push with next commit pointing to the merged ADP PR